src/pocketmine/item/Item.php

```diff
@@ -655,6 +655,14 @@ public function getFuelTime() : int{
 		return 0;
 	}
 
+	/**
+	 * Returns an item after burning fuel
+	 */
+	public function getFuelResidue() : Item{
+		$this->pop();
```

Comment:
This will produce unexpected behaviour when called multiple times. It should clone itself first.

Reply:
I also thought about it at first, but then I forgot XD